### PR TITLE
time: Fix the Windows local_timezone column value

### PR DESCRIPTION
**Pull request by alessandrogario**
### Without the patch

**Timezone configuration**

```
Z:\osquery>tzutil /g
GMT Standard Time
```

**Without the TZ environment variable**

```
Z:\osquery> osqueryd.exe -S "SELECT timestamp, datetime, timezone, local_timezone FROM time;"
+------------------------------+----------------------+----------+----------------+
| timestamp                    | datetime             | timezone | local_timezone |
+------------------------------+----------------------+----------+----------------+
| Mon Jun 27 13:33:45 2022 UTC | 2022-06-27T13:33:45Z | UTC      | UTC            |
+------------------------------+----------------------+----------+----------------+
```

**With the TZ environment variable**

The environment variable is applying but osquery is not able to parse it correctly

```
Z:\osquery>set TZ=IST-5:30
```

```
osqueryd.exe -S "SELECT timestamp, datetime, timezone, local_timezone FROM time;"
+------------------------------+----------------------+----------+----------------+
| timestamp                    | datetime             | timezone | local_timezone |
+------------------------------+----------------------+----------+----------------+
| Mon Jun 27 13:39:55 2022 UTC | 2022-06-27T13:39:55Z | UTC      | IST            |
+------------------------------+----------------------+----------+----------------+
```

### With the patch applied

**Timezone configuration**

```
Z:\osquery>tzutil /g
GMT Standard Time
```

**Without the TZ environment variable**

```
Z:\osquery> osqueryd.exe -S "SELECT timestamp, datetime, timezone, local_timezone FROM time;"
+------------------------------+----------------------+----------+-------------------+
| timestamp                    | datetime             | timezone | local_timezone    |
+------------------------------+----------------------+----------+-------------------+
| Mon Jun 27 13:47:05 2022 UTC | 2022-06-27T13:47:05Z | UTC      | GMT Standard Time |
+------------------------------+----------------------+----------+-------------------+
```

**With the TZ environment variable**

The environment variable is completely ignored when using the Windows API

```
Z:\osquery>set TZ=IST-5:30
```

```
osqueryd.exe -S "SELECT timestamp, datetime, timezone, local_timezone FROM time;"
+------------------------------+----------------------+----------+-------------------+
| timestamp                    | datetime             | timezone | local_timezone    |
+------------------------------+----------------------+----------+-------------------+
| Mon Jun 27 13:47:31 2022 UTC | 2022-06-27T13:47:31Z | UTC      | GMT Standard Time |
+------------------------------+----------------------+----------+-------------------+
```

**Additional examples**

```
Z:\osquery>tzutil /g
FLE Standard Time

Z:\osquery>osqueryd.exe -S "SELECT timestamp, datetime, timezone, local_timezone FROM time;"
+------------------------------+----------------------+----------+-------------------+
| timestamp                    | datetime             | timezone | local_timezone    |
+------------------------------+----------------------+----------+-------------------+
| Mon Jun 27 13:09:18 2022 UTC | 2022-06-27T13:09:18Z | UTC      | FLE Standard Time |
+------------------------------+----------------------+----------+-------------------+
```

```
Z:\osquery>tzutil /g
Tokyo Standard Time

Z:\osquery>osqueryd.exe -S "SELECT timestamp, datetime, timezone, local_timezone FROM time;"
+------------------------------+----------------------+----------+---------------------+
| timestamp                    | datetime             | timezone | local_timezone      |
+------------------------------+----------------------+----------+---------------------+
| Mon Jun 27 13:12:15 2022 UTC | 2022-06-27T13:12:15Z | UTC      | Tokyo Standard Time |
+------------------------------+----------------------+----------+---------------------+
```

### CODE REVIEW

**Comment by directionless** [comment]
I can't quite tell, what's this fixing? That without a timezone `local_timezone` is UTC when it should be GMT?

**Review by directionless — APPROVED** [review]
Ah, okay! I think some of your examples may be swapped, but it's a bit hard to follow. Too much mostly identical info

**Comment by Smjert** [comment]
@directionless  another way to say this is that the version **before this fix**, which is using the POSIX APIs, was either not showing the timezone at all (falling back to UTC, in the case of the timezone set via Windows interface), or it was taking the timezone from the TZ environment variable, but was only showing the prefix with no offset (IST only vs IST-5:30).

**After the fix**, it's using the Windows APIs to get the timezone information; these are only reacting to the timezone set via Windows interface (UI or otherwise) and it's showing it in the long format, which doesn't have collisions (like AMST, which either means America Summer Time or Armenia Summer Time, which have definitely different offsets).
